connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -0,0 +1,628 @@
+
+/**
+ * Catalog interface for Spark. To access this, use `SparkSession.catalog`.
+ *
+ * @since 3.4.0
+ */
+abstract class Catalog {
+
+  /**
+   * Returns the current database (namespace) in this session.
+   *
+   * @since 3.4.0
+   */
+  def currentDatabase: String
+
+  /**
+   * Sets the current database (namespace) in this session.
+   *
+   * @since 3.4.0
+   */
+  def setCurrentDatabase(dbName: String): Unit
+
+  /**
+   * Returns a list of databases (namespaces) available within the current catalog.
+   *
+   * @since 3.4.0
+   */
+  def listDatabases(): Dataset[Database]
+
+  /**
+   * Returns a list of tables/views in the current database (namespace). This includes all
+   * temporary views.
+   *
+   * @since 3.4.0
+   */
+  def listTables(): Dataset[Table]
+
+  /**
+   * Returns a list of tables/views in the specified database (namespace) (the name can be
+   * qualified with catalog). This includes all temporary views.
+   *
+   * @since 3.4.0
+   */
+  @throws[AnalysisException]("database does not exist")
+  def listTables(dbName: String): Dataset[Table]
+
+  /**
+   * Returns a list of functions registered in the current database (namespace). This includes all
+   * temporary functions.
+   *
+   * @since 3.4.0
+   */
+  def listFunctions(): Dataset[Function]
+
+  /**
+   * Returns a list of functions registered in the specified database (namespace) (the name can be
+   * qualified with catalog). This includes all built-in and temporary functions.
+   *
+   * @since 3.4.0
+   */
+  @throws[AnalysisException]("database does not exist")
+  def listFunctions(dbName: String): Dataset[Function]
+
+  /**
+   * Returns a list of columns for the given table/view or temporary view.
+   *
+   * @param tableName
+   *   is either a qualified or unqualified name that designates a table/view. It follows the same
+   *   resolution rule with SQL: search for temp views first then table/views in the current
+   *   database (namespace).
+   * @since 3.4.0
+   */
+  @throws[AnalysisException]("table does not exist")
+  def listColumns(tableName: String): Dataset[Column]
+
+  /**
+   * Returns a list of columns for the given table/view in the specified database under the Hive
+   * Metastore.
+   *
+   * To list columns for table/view in other catalogs, please use `listColumns(tableName)` with
+   * qualified table/view name instead.
+   *
+   * @param dbName
+   *   is an unqualified name that designates a database.
+   * @param tableName
+   *   is an unqualified name that designates a table/view.
+   * @since 3.4.0
+   */
+  @throws[AnalysisException]("database or table does not exist")
+  def listColumns(dbName: String, tableName: String): Dataset[Column]
+
+  /**
+   * Get the database (namespace) with the specified name (can be qualified with catalog). This
+   * throws an AnalysisException when the database (namespace) cannot be found.
+   *
+   * @since 3.4.0
+   */
+  @throws[AnalysisException]("database does not exist")
```

Comment:
`@throws[AnalysisException]` cannot be removed, otherwise it will be judged as incompatible.

Reply:
Yeah we will need to do something about this in a bit. It is fine for now.

Reply:
ok